### DISCUSSION TITLE
Create a role based on an existing role

### DISCUSSION THREAD

**Discussion by timnick-snow** [apache/superset#18845] [Ideas]
Creating a role is annoying because there are too many permissions。
so is it possible to create a new role based on the permissions of the existing role.
For example, subtract some permissions from Gamma to get a custom role. Keep the Gamma role the same

This may not be difficult to implement, but it should be helpful for creating new characters

**Comment by LalaGabor** [comment]
100% support this.....extraordinarily annoying

**Comment by villebro** [comment]
Can't you have a base role and then create additional roles that supplement the original, and then just add the user to both roles?

**Comment by timnick-snow** [comment]
> Can't you have a base role and then create additional roles that supplement the original, and then just add the user to both roles?



This is easy to implement when you want to add permissions based on the Gamma role. However, it is very difficult to remove some permissions

Creating basic characters is difficult and tedious, and I prefer to build based on already provided characters such as Gamma.

**Comment by villebro** [comment]
Converting this to a discussion, as this isn't a bug, but rather a FAB feature request.
